src/model/preemption.c

```diff
@@ -33,7 +33,7 @@ exception_t preemptionPoint(void)
 #ifdef CONFIG_KERNEL_MCS
         } else {
             updateTimestamp();
-            if (!checkBudget()) {
+            if (!(sc_active(NODE_STATE(ksCurSC)) && isSufficientAndSplittable()) || isCurDomainExpired()) {
```

Comment:
If ksCurSC has been revoked just before this, then `sc->scRefillMax = 0;` and so there's no capacity in the SC for the new ksConsumed to be subtracted from.  Does ksConsumed just get added to the next thread that gets picked to run?

Reply:
The operation would not have commenced if the current SC didn't have enough budget and the if the operation produced an overrun the SC would have been charged the excess. Simply losing the ksConsumed would be equivalent to charging the current SC then clearing it anyway.

I think it would make sense to add a branch to `mcsIRQ` that sets the `ksConsumed` to 0 if the current SC is no longer configured.

* [ ] Add commit to set `ksConsumed` to 0 if no `ksCurSC` is no longer configured